code/integration/main/main.ino

```diff
@@ -56,33 +62,42 @@ void loop()
       String received_string = myWiFiService.Read();                    // get string from Wifly
       g_states = myParser.RunParser(received_string, 0, 0);             // interpret string
       String parser_return_string = myParser.Get_String_from_Parser();  // get string for factory
-      
-      switch ( g_states )                                               // evaluate next steps
+
+      switch ( g_states )
       {
-        case ERROR_STATE:
-          break;
-        case LOGIN_SUCCESSFUL:
-          break;
-        case LOGIN_PW_WRONG:
-          break;
-        case LOGOUT_SUCCESSFUL:
+        case ERROR_STATE:                           // error states have same result
+        case LOGIN_PW_WRONG:                        // error counter ++
+        case LOGOUT_PW_WRONG:                        
+        case ORDER_WRONG:                           
+        case ORDER_PW_WRONG:                        
+          g_error_count++;
+        case LOGIN_SUCCESSFUL:                      // client detected
+          myAuftragsverwaltung.NewClientDetected();
           break;
-        case LOGOUT_PW_WRONG:
+        case LOGOUT_SUCCESSFUL:                     // do nothing?
           break;
-        case ORDER_SUCCESSFUL:
+        case ORDER_SUCCESSFUL:                      // initiate order
+          myAuftragsverwaltung.NewOrderRegistered( parser_return_string, 0); // TODO: where do we get the time from (set to 0 now) ?
           break;
-        case ORDER_WRONG:
+        case BROADCAST:                             // dunno
           break;
-        case ORDER_PW_WRONG:
+        case CLIENT_CONNECT:                        // do nothing, pw not set yet
           break;
-        case BROADCAST:
-          break;
-        case CLIENT_CONNECT:
-          break;
-        case CLIENT_DISCONNECT:
+        case CLIENT_DISCONNECT:                     // *CLOS* detected -> client left network
+          g_error_count = 0;                        // reset error count (should be in Auftragsverwaltung)
           break;
         default:
-          break;
+          break;              
+        }
+        
+      }
+
+      myWiFiService.Send( parser_return_string );                       // send answer back to client      
+      
+      if( g_error_count > MAX_ERROR_COUNT_SESSION)                      // check whether error count exceeds limit
```

Comment:
Dies ist ein recht einfacher Fehler-detektier-mechanismus. Einfach im Sinne von simpel. In einer unsererer Vorherigen Diskussionen haben wir gesagt, dass es es sinnvoll ist wenn wir nur falsche Nachrichten innerhalb von einem bestimmten Zeitfenster betrachten und dann nach einer gewissen Zeit den Fehler-Counter zurueck setzen. Ein (1) weiterer Timer ist dafuer notwendig.
In diesem simpelen Ansatz gibt es nur einen maximalen Fehler-Count pro Session. Dieser wird zurueck auf null (0) gesetzt, wenn ein Client die Session beendet. Wenn zu viele falsche Nachrichten gesendet werden, wird der Client aus dem Network gekickt (wobei das ja noch nicht funktioniert) und die Kommunikation fuer xxx sekunden ausgesetzt. Ich glaube, dass dies ausreichend ist da ein verbundener Client unser Protokoll kennen sollte und ich nicht erwarte, dass zu viele falsche Nachrichten gesendet werden. Was meint ihr? Mit oder ohne timer?